### PR TITLE
feat(core): Make OAuth2 error handling consistent with success handling

### PR DESCRIPTION
**Pull request by ManishDhanwal07**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

**Comment by netroy** [comment]
isn't this a duplicate of https://github.com/n8n-io/n8n/pull/5492 ?

File: packages/cli/src/credentials/oauth2Credential.api.ts [hunk 1]
```diff
@@ -130,7 +131,7 @@ oauth2CredentialController.get(
 			state: stateEncodedStr,
 		};
 
-		await Container.get(ExternalHooks).run('oauth2.authenticate', [oAuthOptions]);
```
Comment (netroy) [inline packages/cli/src/credentials/oauth2Credential.api.ts:133]: any reasons to change this line?

Comment (ManishDhanwal07) [reply packages/cli/src/credentials/oauth2Credential.api.ts]: Hi @netroy 
I have changed it. please have look.

File: packages/cli/templates/oauth-error-callback.handlebars [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<html>
+	<head>
+		<title>n8n - OAuth Callback</title>
+		<style>
+			body { font-family: 'Open Sans', sans-serif; padding: 10px;}
+			pre.error { background: #f7f7f7; border: 1px solid #ddd; border-radius: 3px; padding: 10px; overflow: auto; overflow-wrap: break-word; white-space: pre-wrap; }
+		</style>
+	</head>
+	<body>
+		Failed to connect. The window can be closed now.
```
Comment (krynble) [inline packages/cli/templates/oauth-error-callback.handlebars:10]: I'd recommend moving the sentence `The window can be closed now` to the bottom (after the `<pre>` block) and add a note stating that the error is outlined above, the user can use this information to fix the issue and retry.

**Review by krynble — DISMISSED** [review]
Feature-wise, it seems all right, I just left a small suggestion about wording to clarify what the user can do once facing this issue.

File: packages/cli/templates/oauth-error-callback.handlebars (hunk 2 above)
Comment (ManishDhanwal07) [reply packages/cli/templates/oauth-error-callback.handlebars]: @krynble I have changed the position. thanks

Comment (ManishDhanwal07) [reply packages/cli/templates/oauth-error-callback.handlebars]: Hello @krynble  @netroy 
Is there anything else you think could be done here? if not, Can you please approve the PR?
Thank you.

**Review by netroy — APPROVED** [review]

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.222.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.222.0)

**Pull request merged**